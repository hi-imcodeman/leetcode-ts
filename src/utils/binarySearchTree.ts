import { TreeNode } from './binaryTree'

export function insertLevelOrderBST(arr: number[]|string[]): TreeNode<number|string> | null {
    let root:TreeNode<number|string>|null = null
    const insert = (value:number|string,node:TreeNode<number|string>|null)=>{
        if(node === null)
            return new TreeNode(value)
        if(value < node.val){
            node.left = insert(value,node.left)
        }else if(value >= node.val){
            node.right = insert(value,node.right)
        }
        return node
    }
    arr.forEach((item:number|string)=>{
        root = insert(item,root)
    })
    return root
}