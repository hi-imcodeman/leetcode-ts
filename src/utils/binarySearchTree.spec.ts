import { insertLevelOrderBST } from './binarySearchTree'

describe('Binary Search Tree',()=>{
    test('insertLevelOrderBST:number',()=>{
        const bst = insertLevelOrderBST([1,2,3,4])
        expect(bst).toMatchSnapshot()
    })
    test('insertLevelOrderBST:string',()=>{
        const bst = insertLevelOrderBST(['a','c','A'])
        expect(bst).toMatchSnapshot()
    })
})
