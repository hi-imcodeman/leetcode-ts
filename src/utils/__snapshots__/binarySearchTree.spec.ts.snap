// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Binary Search Tree insertLevelOrderBST:number 1`] = `
TreeNode {
  "left": null,
  "right": TreeNode {
    "left": null,
    "right": TreeNode {
      "left": null,
      "right": TreeNode {
        "left": null,
        "right": null,
        "val": 4,
      },
      "val": 3,
    },
    "val": 2,
  },
  "val": 1,
}
`;

exports[`Binary Search Tree insertLevelOrderBST:string 1`] = `
TreeNode {
  "left": TreeNode {
    "left": null,
    "right": null,
    "val": "A",
  },
  "right": TreeNode {
    "left": null,
    "right": null,
    "val": "c",
  },
  "val": "a",
}
`;
